src/components/InternalOverlay/index.js

```diff
@@ -95,9 +95,10 @@ const InternalOverlay = props => {
         if (isVisible && contentMeta) {
             onOpened();
             disableBodyScroll(undefined, { reserveScrollBarGap: true });
-        } else {
-            enableBodyScroll();
         }
+        return () => {
+            enableBodyScroll();
+        };
```

Comment:
I have doubts, whats happen when set isVisible to false and the component is not unmounted? then enableBodyScroll will not be called when it should do.

Reply:
React also sanitizes the effects of previous renderings before running the effects of the current render. So when you change isVisible enableBodyScroll is always called

https://es.reactjs.org/docs/hooks-effect.html#example-using-hooks-1